docs/user-guide/install.md

```diff
@@ -0,0 +1,307 @@
+
+### Docker Runtime Configuration
+
+During installation, the Sysbox installer will reconfigure the Docker daemon such
+that it detects the Sysbox runtime. It does this by adding the following
+configuration to `/etc/docker/daemon.json` and sending a signal (SIGHUP) to Docker:
+
+```json
+{
+   "runtimes": {
+      "sysbox-runc": {
+         "path": "/usr/local/sbin/sysbox-runc"
+      }
+   }
+}
+```
+
+If all is well, Docker will recognize the Sysbox runtime:
+
+```console
+$ docker info | grep -i runtime
+WARNING: No swap limit support
+ Runtimes: runc sysbox-runc
+  Default Runtime: runc
+```
+
+### Docker Userns-Remap
+
+During installation, the Sysbox installer will also check if Docker needs to be placed
+in [userns-remap mode](https://docs.docker.com/engine/security/userns-remap/) or not.
+
+The installer uses the following logic:
+
+-   If the kernel carries a module called `shiftfs`, then Docker does not need to be
+    placed in userns-remap mode.
+
+    -   This is normally the case for the Ubuntu desktop and server editions.
+
+-   Otherwise, Docker does need to be placed in userns-remap mode.
+
+    -   This is likely the case for some Ubuntu cloud editions.
+
+If Docker needs to be placed in userns-remap, the Sysbox installer will check if
+Docker is already in this mode (by looking for `userns-remap` in
+`/etc/docker/daemon.json` and `userns` entry in `docker info` output). If so, no further
+action is required.
+
+Otherwise, the Sysbox installer will ask the user for permission to transition Docker
+into `userns-remap` mode. If user responds affirmatively, Sysbox installer will make
+the required changes (below) and restart Docker automatically. Alternatively, the user
+will need to make these changes manually and restart Docker service afterwards (e.g.,
+`systemctl restart docker`)
+
+```console
+{
+   "userns-remap": "sysbox",
+}
+```
+
+When Docker is placed in userns-remap mode, there are a couple of caveats to
+keep in mind:
+
+-   Configuring Docker this way places a few functional limitations on regular
+    Docker containers (those launched with Docker's default `runc`), as described
+    in this [Docker document](https://docs.docker.com/engine/security/userns-remap).
+
+-   System container isolation, while strong, is reduced compared to using
+    `shiftfs`. That's because userns-remap causes Docker to put Sysbox into
+    "Directed Userns ID Mapping" mode. See [here](security.md#user-namespace-id-mapping)
+    for more info on this.
+
+## Uninstalling Sysbox
+
+Prior to uninstalling Sysbox, make sure all system containers are removed.
+There is a simple shell script to do this [here](../../scr/rm_all_syscont).
+
+1) Uninstall Sysbox binaries plus all the associated configuration and Systemd
+files:
+
+```console
+$ sudo apt-get purge sysbox -y
+```
+
+2) Remove the `sysbox` user from the system:
+
+```console
+$ sudo userdel sysbox
+```
+
+## Hitless Installation
```

Comment:
I would keep it simple and avoid this section as it can confuse people. I think the subset of people that will have an issue with stopping containers prior to installing Sysbox is very small. This section can hurt more than it helps.

Reply:
Thanks for the review. Will make some adjustments as discussed.